Language/Expressions/Property_Extraction/Instance_Method_Closurization/method_closurization_named_params_A04_t01.dart

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Let `o` be an object, and let `u` be a fresh final variable bound
+/// to o. The closurization of method `f` on object `o` is defined to be
+/// equivalent to:
+/// ```
+/// - <X1 extends B′1, ..., Xs extends B′s>
+///     (T1 p1, ..., Tn pn, {Tn+1 pn+1 = d1, ..., Tn+k pn+k = dk}) =>
+///       u.m<X1, ..., Xs>(p1, ..., pn, pn+1: pn+1, ..., pn+k: pn+k);
+///  ```
+///  where `f` is an instance method named `m` which has type parameter
+///  declarations `X1 extends B1, ..., Xs extends Bs`, required parameters
+///  `p1, ..., pn`, and named parameters `pn+1, ..., pn+k` with defaults
+///  `d1, ..., dk`, using `null` for parameters whose default value is not
+///  specified.
+///  ...
+///  The parameter types `Tj,j ∈ 1..n + k`, are determined as follows: Let the
+///  method declaration `D` be the implementation of `m` which is invoked by the
+///  expression in the body. Let `T` be the class that contains D.
+///  ...
+///  If `T` is a non-generic class then for `j ∈ 1..n+k, Tj` is a type
+///  annotation that denotes the same type as that which is denoted by the type
+///  annotation on the corresponding parameter declaration in `D`. If that
+///  parameter declaration has no type annotation then `Tj` is `dynamic`.
+///
+/// @description Check that if `T` is a non-generic class then for
```

Comment:
But `C` is a generic class. We need a test for both cases, so I'd suggest that the type parameter declaration is removed from the declaration of `C` and the parameter `r1` gets some other type (`String` is fine, because we don't have any other positional parameters whose type is a plain constant type).

Reply:
Ah, now I see. I thought that `T` is a type argument but missed that `Let T be the class that contains D.`. Then, of course, `C` should be a non-generic to follow the description. Fixed. Thank you for pointing.

Reply:
Thanks!